sklearn/tests/test_isotonic.py

```diff
@@ -346,3 +346,30 @@ def test_isotonic_zero_weight_loop():
 
     # This will hang in failure case.
     regression.fit(x, y, sample_weight=w)
+
+
+def test_fast_predict():
+    # test that the faster prediction (https://github.com/scikit-learn/scikit-learn/pull/6206)
+    # change doesn't affect out-of-sample predictions.
+    np.random.seed(123)
```

Comment:
Please do not seed the global numpy singleton to avoid side effects. Instead do:

```python
rng = np.random.RandomState(123)
```

Then replaces occurrences of `np.random.random` with `rng.random(N)`